testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -886,7 +886,7 @@
   next if has_select?(field, selected: option)
 
   # Custom React selector
-  next if has_xpath?("//*[contains(@class, 'class-#{field}__value-container')]/*[contains(text(),'#{option}')]")
+  next if has_xpath?("//*[contains(@class, 'data-testid-#{field}-child__value-container')]/*[contains(text(),'#{option}')]")
```

Comment:
If I'm not wrong, what Capybara will expect here is something like `has_xpath?("//*[contains(@data-testid, 'xxxx')]")`

Reply:
I don't think that's correct, we're not simply looking for `data-testid` since selects use popouts which are moved in the regular DOM tree. To explain another way, the elements we're looking for might not be direct children of the element with the test id since they're detached to allow select dropdowns to work despite scroll overflow etc.

Reply:
Who said that that was the purpose of this improvement? This PR is a fix for the test suite error and adds `data-testid` to the frontend component while keeping backwards compatibility. That's what's also in the PR description. It is not the point of this PR to go and retroactively update all existing tests to use `data-testid`. If you want to add the required Capybara configuration, go over the affected tests, etc feel free.

Reply:
In other words, in product code, looking this line:
```
  const classNamePrefix = testClassName ? `class-${testClassName}-child` : undefined;
```
What prevents us to have
```
  const classNamePrefix = testClassName ? `data-testid-${testClassName}__value` : undefined;
```
So we avoid changes in test side and then is transparent? And then we don't even need to setup the data-testid Capybara configuration.

Reply:
The `class-` prefix doesn't say anything about tests and since it doesn't show up as used code, looks safe to remove if you don't magically guess it's used by tests. The whole concept of using test ids etc is to be explicit that one binding or another is for testing purposes.

Reply:
`data-testid` is present in the solution as well, the code you're quoting is for the children of that parent, hence the classname `data-testid-foo-child`. It doesn't conflict with `data-testid` in any way, namely in that it isn't even an id.

Reply:
Fair enough. Go with your proposal. Just let it work and make it green. Fine for now ;)